internal/config/projectconfig/init.go

```diff
@@ -47,15 +30,45 @@ func SetRootDir(dir string) {
 }
 
 func Init(dir string, projectName string, projectContext *ZeroProjectConfig) {
```

Comment:
Might be worth refactoring this a bit.. Maybe rename this to `CreateProjectConfigFile`? 
And then since `GetProjectFileContent` probably isn't useful on its own, maybe merge the code into this function and then change the test to test this function and assert that the file was created and contains the proper content..
Right now you exposed `GetProjectFileContent` but it's not useful outside the package, and the test is testing internal functionality that was exposed only to be able to test it. 
An external test (in a `_test` file) should be testing only the exposed API of the package, and a package shouldn't expose more than it needs to.
Looks like you have a TODO for that below, but I would say combining these functions and then having a single test would be the way to go.